drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/KieFileSystemImpl.java

```
@@ -15,118 +15,162 @@
         return mfs;
     }
 
-    public KieFileSystem generateAndWritePomXML(ReleaseId releaseId) {
-        write("pom.xml", KieBuilderImpl.generatePomXml(releaseId) );
+    public KieFileSystem generateAndWritePomXML( ReleaseId releaseId ) {
+        write( "pom.xml", KieBuilderImpl.generatePomXml( releaseId ) );
         return this;
     }
 
-    public KieFileSystem writePomXML(byte[] content) {
-        write("pom.xml", content);
+    public KieFileSystem writePomXML( byte[] content ) {
+        write( "pom.xml", content );
         return this;
     }
 
-    public KieFileSystem writePomXML(String content) {
-        write("pom.xml", content);
+    public KieFileSystem writePomXML( String content ) {
+        write( "pom.xml", content );
         return this;
     }
 
-    public KieFileSystem writeKModuleXML(byte[] content) {
-        write(KieModuleModelImpl.KMODULE_SRC_PATH, content);
+    public KieFileSystem writeKModuleXML( byte[] content ) {
+        write( KieModuleModelImpl.KMODULE_SRC_PATH, content );
         return this;
     }
 
-    public KieFileSystem writeKModuleXML(String content) {
-        write(KieModuleModelImpl.KMODULE_SRC_PATH, content);
+    public KieFileSystem writeKModuleXML( String content ) {
+        write( KieModuleModelImpl.KMODULE_SRC_PATH, content );
         return this;
     }
-    
+
+    public MemoryFileSystem getMfs() {
+        return mfs;
+    }
+
+    public KieFileSystem clone() {
+        try {
+            final ByteArrayOutputStream byteArray = writeToByteArray( this );
+            final KieFileSystem kieFileSystem = readFromByteArray( byteArray );
+
+            return kieFileSystem;
+        } catch ( IOException ioe ) {
+            logger.error( "Error cloning KieFileSystemImpl: " + this.toString(), ioe );
+            return null;
+        } catch ( ClassNotFoundException cnfe ) {
+            logger.error( "Error cloning KieFileSystemImpl: " + this.toString(), cnfe );
```

Sorry for being nosy :) -- should we fail fast in this case? Logging the exception as an error suggests the state is not recoverable. But the method actually returns "normally".